boa_engine/src/native_function.rs

```diff
@@ -197,169 +195,3 @@ impl NativeFunction {
-        Self {
-            inner: GenericInner::PointerFn(function),
-        }
-    }
-
-    /// Creates a `GenericNativeFunction` from a `Copy` closure.
-    pub fn from_copy_closure<F>(closure: F) -> Self
-    where
-        F: FnMut(Args, &mut Context<'_>) -> Ret + Copy + 'static,
-    {
-        // SAFETY: The `Copy` bound ensures there are no traceable types inside the closure.
-        unsafe { Self::from_closure(closure) }
-    }
-
-    /// Creates a `GenericNativeFunction` from a `Copy` closure and a list of traceable captures.
-    pub fn from_copy_closure_with_captures<F, T>(closure: F, captures: T) -> Self
-    where
-        F: FnMut(Args, &mut T, &mut Context<'_>) -> Ret + Copy + 'static,
-        T: Trace + 'static,
-    {
-        // SAFETY: The `Copy` bound ensures there are no traceable types inside the closure.
-        unsafe { Self::from_closure_with_captures(closure, captures) }
-    }
-
-    /// Creates a new `GenericNativeFunction` from a closure.
-    ///
-    /// # Safety
-    ///
-    /// Passing a closure that contains a captured variable that needs to be traced by the garbage
-    /// collector could cause an use after free, memory corruption or other kinds of **Undefined
-    /// Behaviour**. See <https://github.com/Manishearth/rust-gc/issues/50> for a technical explanation
-    /// on why that is the case.
-    pub unsafe fn from_closure<F>(mut closure: F) -> Self
-    where
-        F: FnMut(Args, &mut Context<'_>) -> Ret + 'static,
-    {
-        // SAFETY: The caller must ensure the invariants of the closure hold.
-        unsafe {
-            Self::from_closure_with_captures(move |args, _, context| closure(args, context), ())
-        }
-    }
-
-    /// Create a new `GenericNativeFunction` from a closure and a list of traceable captures.
-    ///
-    /// # Safety
-    ///
-    /// Passing a closure that contains a captured variable that needs to be traced by the garbage
-    /// collector could cause an use after free, memory corruption or other kinds of **Undefined
-    /// Behaviour**. See <https://github.com/Manishearth/rust-gc/issues/50> for a technical explanation
-    /// on why that is the case.
-    pub unsafe fn from_closure_with_captures<F, T>(closure: F, captures: T) -> Self
-    where
-        F: FnMut(Args, &mut T, &mut Context<'_>) -> Ret + 'static,
-        T: Trace + 'static,
-    {
-        Self {
-            inner: GenericInner::Closure(Box::new(GenericClosure {
-                f: closure,
-                captures,
-                phantom: PhantomData,
-            })),
-        }
-    }
-
-    /// Calls this `GenericNativeFunction`, forwarding the arguments to the corresponding function.
-    #[inline]
-    pub fn call(&mut self, args: Args, context: &mut Context<'_>) -> Ret {
-        match self.inner {
-            GenericInner::PointerFn(f) => f(args, context),
-            GenericInner::Closure(ref mut c) => c.call(args, context),
-        }
-    }
-}
```

Comment:
I think removing this is not desired, right?

Reply:
Yep. I initially added this to implement the callbacks for the hooks, but while designing I found out that we could use a trait object with the implementations instead, so we don't need this anymore.